### DISCUSSION TITLE
QSPI NOR Flash support

### DISCUSSION THREAD

**Discussion by andrenemat** [polarfire-soc/polarfire-soc-documentation#105] [HSS and U-Boot] [answered]
Do we support booting the Polarfire SoC from QSPI NOR Flash, and if so are there any approved and tested parts?

**Comment by vfalanis** [comment, marked as answer]
Hi @andrenemat ,

We currently only support booting from QSPI NAND Flash only, specifically W25N01GVZEIG/IT. This is documented and described [here](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/boards/mpfs-icicle-kit-es/booting-from-qspi/booting-from-qspi.md)

**Reply by vfalanis** [reply]
The HSS provides an MSS QSPI driver and has hooks in place to add support for more NAND or NOR flash devices if required. Customers could use the existing infrastructure provided to add support for the flash device of their choice. For more information please refer to the HSS qspi service API [here](https://github.com/polarfire-soc/hart-software-services/tree/9ba61c3df3743c191b4672b110e875b4b74b427b/services/qspi)
